pkg/controller/deployment/deployment_controller.go

```diff
@@ -417,6 +420,13 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		dc.enqueueDeployment(&d)
 		return nil
 	}
+	if !dc.podStoreSynced() {
+		// Sleep so we give the pod reflector goroutine a chance to run.
+		time.Sleep(PodStoreSyncedPollPeriod)
```

Comment:
Is there a reason we need 2 separate sync intervals? Why not:
```
 	if !dc.rsStoreSynced() || !dc.podStoreSynced() {
 		// Sleep so we give the reflector goroutine a chance to run.
 		time.Sleep(StoreSyncedPollPeriod)
 		glog.Infof("Waiting for store to sync, requeuing deployment %s", d.Name)
  		dc.enqueueDeployment(&d)
  		return nil
  	}
```

And where is the deployment store checked?